app/views/addresses/_form.html.erb

```diff
@@ -33,4 +37,57 @@
   <% end %>
 </div>
 
-</div>
+<script>
+  function initMap() {
+    const map = new google.maps.Map(document.getElementById("map"), {
+      center: { lat: 19.2433, lng: -103.725 }, // Centra el mapa en la ciudad de Colima
+      zoom: 14,
+    });
+
+    const input = document.getElementById("address-input");
+    const autocomplete = new google.maps.places.Autocomplete(input);
+
+    map.addListener("click", function(event) {
+      const geocoder = new google.maps.Geocoder();
+      geocoder.geocode({ location: event.latLng }, function(results, status) {
+        if (status === "OK" && results[0]) {
+          input.value = results[0].formatted_address;
+          fillAddressFields(results[0]);
+        }
+      });
+    });
+
+    autocomplete.addListener("place_changed", function () {
+      const place = autocomplete.getPlace();
+      if (place.geometry) {
+        map.setCenter(place.geometry.location);
+        fillAddressFields(place);
+      }
+    });
+
+    function fillAddressFields(place) {
+      const addressComponents = place.address_components;
+
+      for (const component of addressComponents) {
+        const types = component.types;
+
+        if (types.includes("locality")) { // Ciudad
+          document.getElementById("address_city").value = component.long_name;
+        } else if (types.includes("administrative_area_level_1")) { // Estado
+          document.getElementById("address_state").value = component.long_name;
+        } else if (types.includes("postal_code")) { // Código Postal
+          document.getElementById("address_zip_code").value = component.long_name;
+        }
+      }
+
+      // Extraer la parte de la dirección correspondiente a la calle, número y colonia
+      const streetNumber = place.address_components.find(component => component.types.includes("street_number"));
+      const route = place.address_components.find(component => component.types.includes("route"));
+      const neighborhood = place.address_components.find(component => component.types.includes("neighborhood"));
+      const addressValue = `${route ? route.long_name : ''}${streetNumber ? ' ' + streetNumber.long_name : ''}${neighborhood ? ', ' + neighborhood.long_name : ''}`;
+      document.getElementById("address-input").value = addressValue;
+    }
+  }
+</script>
```

Comment:
You shouldnt do javascript like this. Use stimulus js controllers and add them to an element, you can also pass data back and forward. Heres a [video example](https://www.youtube.com/watch?v=nf88pbG1a6s) and [another one ](https://www.youtube.com/watch?v=pjYwjOShp0c) and a [blog ](https://dev.to/briankephart/passing-arguments-to-stimulusjs-controllers-1kgi)